### PR TITLE
Testing Math Magicians

### PR DESCRIPTION
**Pull request by yersel500**
## We implement the testing part for Math Magicians Project:

- Write unit tests for the files operate.js and calculate.js using Jest.
- Create unit tests for all React components:
  - Use Jest snapshots to test the components.
  - Use React Testing Library to simulate user interaction.
- Run your full tests suite using Jest. All tests should pass.

### CODE REVIEW

File: src/logic/operate.test.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  });
+
+  test('Third test operate', () => {
+    expect(operate('10', '2', '÷')).toBe('5');
+  });
+
+  test('Fourth test operate', () => {
+    expect(operate('9', '3', 'x')).toBe('27');
+  });
+});
```
Comment (Gambit142) [inline src/logic/operate.test.js]: - [x] Nice work testing most of the functionalities here. If you remember, one of the project requirements states that you should test all functionalities. Kindly add a test example for the _**modulus operator**_ `(%)`. 👍

File: src/pages/Home.test.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+import renderer from 'react-test-renderer';
+import Home from './Home';
+
+it('Home page should render correctly', () => {
+  const tree = renderer.create(<Home />).toJSON();
+  expect(tree).toMatchSnapshot();
+});
```
Comment (Gambit142) [inline src/pages/Home.test.js]: - [ ] Nice work adding the Jest Snapshots tests for your React Components. As you also know, one of the major requirements is for you to use React Testing Library to simulate user interactions or verify elements of the components. You can read this article to add simple tests for each component https://www.robinwieruch.de/react-testing-library/

File: src/pages/Quote.test.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+import renderer from 'react-test-renderer';
+import Quote from './Quote';
+
+it('Quote page should render correctly', () => {
+  const tree = renderer.create(<Quote />).toJSON();
+  expect(tree).toMatchSnapshot();
+});
```
Comment (Gambit142) [inline src/pages/Quote.test.js]: - [x] Nice work adding the Jest Snapshots tests for your React Components. As you also know, one of the major requirements is for you to use React Testing Library to simulate user interactions or verify elements of the components. You can read this article to add simple tests for each component https://www.robinwieruch.de/react-testing-library/

File: src/components/Calculator.test.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+import renderer from 'react-test-renderer';
+import Calculator from './Calculator';
+
+it('The calculator renders correctly', () => {
+  const tree = renderer.create(<Calculator />).toJSON();
+  expect(tree).toMatchSnapshot();
+});
```
Comment (Gambit142) [inline src/components/Calculator.test.js]: - [x] Nice work adding the Jest Snapshots tests for your React Components. As you also know, one of the major requirements is for you to use React Testing Library to simulate user interactions or verify elements of the components. You can read this article to add simple tests for each component https://www.robinwieruch.de/react-testing-library/

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @yersel500 @omar25ahmed, 

Good job so far!

### Highlights
- All linters checks passed ✔️ 
- Use of Jest to test pure functions ✔️ 
- Use of Jest Snapshots to create snapshots of React components ✔️ 

However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
- Kindly delete the `App.test.js` file as it is preventing all your tests from passing. Remember you were to delete all files and components that were created with the `Create React App` command 👍 

- _Kindly check the comments under the review for other required changes._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by alexander16108 — APPROVED** [review]
Hi @yersel500 and @omar25ahmed , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**